invisible_cities/reco/trigger_functions_test.py

```diff
@@ -0,0 +1,330 @@
+        val_p  = True
+
+        elements = [evt_id    , pmt_id
+                   ,t_true    , q_true  , w_true, h_true
+                   ,val_q     , val_w   , val_h , val_p
+                   ,wf[start1-1], wf.max()
+                   ,-1        , -1      , -1]
+
+        for i, element in enumerate(elements):
+            assert element == triggers[j][i]
+
+
+@given(evt_id=integers(1, 10), pmt_id=integers(1, 10))
+def test_get_trigger_candidates_multipeak(double_square_wf, evt_id, pmt_id
+                                         ,channel_conf    , trigger_conf):
+    # Check that the multipeak protection discards the trigger if conditions are met.
+    channel_config = channel_conf.copy()
+    trigger_config = trigger_conf.copy()
+
+    wf     = double_square_wf[0]
+    thr    = double_square_wf[1]
+    length = double_square_wf[2]
+    start1 = double_square_wf[3]
+    start2 = double_square_wf[4]
+    channel_config  = channel_conf.copy()
+    channel_config['pulse_valid_ext'] = 0
+    triggers = get_trigger_candidates(wf, evt_id, pmt_id
+                                     ,channel_config, trigger_config
+                                     ,np.zeros(wf.shape), wf.max())
+    for t in triggers:
+        assert t[-6] == True
+
+    ### Second peak characteristics
+    peak   = wf[start2:start2+length+channel_config['pulse_valid_ext']]
+    t_true = start2
+    q_true = peak[:-2].sum()
+    val_q  = in_range(q_true
+                     , channel_config['q_min'], channel_config['q_max']
+                     , left_closed=False, right_closed=False)
+    w_true = len(peak)
+    val_w  = in_range(w_true
+                     , channel_config['time_min'], channel_config['time_max']
+                     , left_closed=True , right_closed=False)
+    h_true = peak.max()
+    val_h  = h_true < channel_config['amp_max']
+    val_p  = True
+
+    # Multipeak condition metretrieve_trigger_information
+    trigger_config['multipeak'] = {'q_min':q_true//2, 'time_min':w_true//2, 'time_after':length+10}
+    triggers = get_trigger_candidates(wf, evt_id, pmt_id
+                                     ,channel_config, trigger_config
+                                     ,np.zeros(wf.shape), wf.max())
+
+    assert triggers[0][-6] == False
+    assert triggers[1][-6] == True
+
+    # Multipeak ondition no longer met
+    trigger_config['multipeak'] = {'q_min':q_true*2, 'time_min':w_true*2, 'time_after':length+10}
+    triggers = get_trigger_candidates(wf, evt_id, pmt_id
+                                     ,channel_config, trigger_config
+                                     ,np.zeros(wf.shape), wf.max())
+
+    for t in triggers:
+        assert t[-6] == True
+
+
+@given(evt_id=integers(0, 3))
+def test_retrieve_trigger_information(double_square_wf, evt_id
+                                     ,channel_conf    , trigger_conf):
+    channel_config = channel_conf.copy()
+    trigger_config = trigger_conf.copy()
```

Comment:
Because I modify the dictionary on several tests and the fixture's scope is set to session. Ideally I would use 'function' but it gave an ugly warning that, while it does not worry me, adds too much noise to pytest:

` invisible_cities/reco/trigger_functions_test.py::test_retrieve_trigger_information
  /home/lior/miniconda/envs/IC-3.7-2020-06-16/lib/python3.7/site-packages/hypothesis/extra/pytestplugin.py:170: HypothesisDeprecationWarning: invisible_cities/reco/trigger_functions_test.py::test_retrieve_trigger_information uses the 'trigger_config' fixture, which is reset between function calls but not between test cases generated by `@given(...)`.  You can change it to a module- or session-scoped fixture if it is safe to reuse; if not we recommend using a context manager inside your test function.  See https://docs.pytest.org/en/latest/fixture.html#sharing-test-data for details on fixture scope.
    note_deprecation(msg % (fx.argname,), since="2020-02-29")`  

In lack of a better approach, and after discussion with @gonzaponte, we have opted to define two functions that provide the modified configuration dicts when needed.